sys/vm/vm_mmap.c

```diff
@@ -707,8 +699,7 @@ kern_mmap_req(struct thread *td, const struct mmap_req *mrp)
 		if (CHERI_SEALABLE_ALIGNMENT(size) > (1UL << PAGE_SHIFT)) {
 			flags |= MAP_ALIGNED(CHERI_SEAL_ALIGN_SHIFT(size));
 
-			if (size != CHERI_SEALABLE_LENGTH(size) &&
-			    (cheriabi_mmap_precise_bounds || (flags & MAP_FIXED))) {
+			if (size != CHERI_SEALABLE_LENGTH(size)) {
```

Comment:
Not necessarily for this PR, but now that we have seemingly abandoned the notion of different alignment granularities for unsealed and sealed caps, could we remove some code?

Reply:
I think the code is small enough that I'm OK with leaving it alone in case someone wants to experiment with difference tradeoffs.  This isn't something we'd upstream as is though.